x-pack/plugins/infra/server/lib/adapters/metrics/kibana_metrics_adapter.ts

```diff
@@ -45,7 +46,7 @@ export class KibanaMetricsAdapter implements InfraMetricsAdapter {
 
     const validNode = await checkValidNode(search, indexPattern, nodeField, options.nodeId);
     if (!validNode) {
-      throw new Error(
+      throw new InvalidNodeError(
```

Comment:
This introduces knowledge about GraphQL into the `libs` part of the codebase. It might not be worth investing the complexity cost of keeping this separation intact though. :thinking:

@skh, do you have any opinion on this?

Reply:
Agreed. The assumptions around our approach at the start of the project have changed. It's getting harder and harder to justify the extra effort to maintain the separation, most of the benefits will probably never be realized.